packages/client-react-streaming/src/ReadableStreamLink.tsx

```diff
@@ -0,0 +1,137 @@
+import { ApolloLink, Observable } from "@apollo/client/index.js";
+import type { FetchResult } from "@apollo/client/index.js";
+
+export type ReadableStreamLinkEvent =
+  | { type: "next"; value: FetchResult }
+  | { type: "completed" }
+  | { type: "error" };
+
+interface InternalContext {
+  [teeToReadableStreamKey]?: ReadableStreamDefaultController<ReadableStreamLinkEvent>;
+  [readFromReadableStreamKey]?: ReadableStream<ReadableStreamLinkEvent>;
+}
+
+const teeToReadableStreamKey = Symbol.for(
+  "apollo.tee.readableStreamController"
+);
+const readFromReadableStreamKey = Symbol.for("apollo.read.readableStream");
+
+/**
+ * Apply to a context that will be passed to a link chain containing `TeeToReadableStreamLink`.
+ * @param controller
+ * @param context
+ * @returns
+ */
+export function teeToReadableStream<T extends Record<string, any>>(
+  controller: ReadableStreamDefaultController<ReadableStreamLinkEvent>,
+  context: T
+): T & InternalContext {
+  return Object.assign(context, {
+    [teeToReadableStreamKey]: controller,
+  });
+}
+
+/**
+ * Apply to a context that will be passed to a link chain containing `ReadFromReadableStreamLink`.
+ * @param readableStream
+ * @param context
+ * @returns
+ */
+export function readFromReadableStream<T extends Record<string, any>>(
+  readableStream: ReadableStream<ReadableStreamLinkEvent>,
+  context: T
+): T & InternalContext {
+  return Object.assign(context, {
+    [readFromReadableStreamKey]: readableStream,
+  });
+}
+
+/**
+ * A link that allows the request to be cloned into a readable stream, e.g. for
+ * transport of multipart responses from RSC or a server loader to the browser.
+ */
+export const TeeToReadableStreamLink = new ApolloLink((operation, forward) => {
```

Comment:
Looking at #389, it looks like these are used internally and not actually set by the user. If that is the case, should we mark these links as `@internal`? (same goes for the read link)

```suggestion
/**
 * A link that allows the request to be cloned into a readable stream, e.g. for
 * transport of multipart responses from RSC or a server loader to the browser.
 *
 * @internal
 */
export const TeeToReadableStreamLink = new ApolloLink((operation, forward) => {
```

Reply:
I'd say that everything in this package could be considered "internal" or as a building block for "build-your-own-framework". These links will e.g. be used outside of the ApolloClient implementations of this package in the React-Router implementation.